app/components/more-details.jsx

```diff
@@ -0,0 +1,146 @@
+// https://github.com/EnCiv/civil-pursuit/issues/89
+
+'use strict'
+import React, { useEffect, useState } from 'react'
+import { createUseStyles } from 'react-jss'
+import cx from 'classnames'
+import { JsonForms } from '@jsonforms/react'
+import { vanillaCells, vanillaRenderers } from '@jsonforms/vanilla-renderers'
+import { PrimaryButton } from './button.jsx'
+import { rankWith, isControl } from '@jsonforms/core'
+import { withJsonFormsControlProps } from '@jsonforms/react'
+
+const CustomInputRenderer = withJsonFormsControlProps(({ data, handleChange, path, uischema, schema, classes }) => {
+  const options = schema.enum || []
+  const label = schema.title || uischema.label
+
+  let type
+  if (schema.format === 'date') {
+    type = 'date'
+  } else if (schema.type === 'integer' || schema.type === 'number') {
+    type = 'number'
+  } else if (schema.type === 'boolean') {
+    type = 'checkbox'
+  } else if (options.length > 0) {
+    type = 'select'
+  } else {
+    type = 'text'
+  }
+
+  return (
+    <div>
+      <label>{label}</label>
+      {type === 'select' ? (
+        <select value={data || ''} onChange={i => handleChange(path, i.target.value)} className={classes.formInput}>
+          <option value="" disabled>
+            Choose one
+          </option>
+          {options.map(option => (
+            <option key={option} value={option}>
+              {option}
+            </option>
+          ))}
+        </select>
+      ) : (
+        <input
+          type={type}
+          checked={type === 'checkbox' ? data : undefined}
+          value={type === 'checkbox' ? undefined : data || ''}
+          onChange={i => handleChange(path, type === 'checkbox' ? i.target.checked : i.target.value)}
+          className={classes.formInput}
+        />
+      )}
+    </div>
+  )
+})
+
+const customRenderers = [...vanillaRenderers, { tester: rankWith(3, isControl), renderer: CustomInputRenderer }]
+
+const MoreDetails = props => {
+  const { className = '', schema = {}, uischema = {}, details = {}, onDone = () => {}, ...otherProps } = props
+  const { title } = otherProps
+  const [data, setData] = useState(details)
+  const [isValid, setIsValid] = useState(false)
+  const classes = useStyles(props)
+
+  const handleOnClick = () => {
+    if (isValid) {
+      onDone({ valid: true, value: data })
+    }
+  }
+
+  const handleIsValid = () => {
+    const requiredData = schema.properties || {}
+    return Object.keys(requiredData).every(i => (requiredData[i].enum ? data[i] !== undefined && data[i] !== '' : true))
+  }
+
+  useEffect(() => {
+    setIsValid(handleIsValid())
+  }, [data, schema])
+
+  return (
+    <div className={cx(classes.formContainer, className)}>
+      {title && <p className={classes.formTitle}>{title}</p>}
+      <div className={classes.jsonFormContainer}>
+        <JsonForms
+          schema={schema}
+          uischema={uischema}
+          data={data}
+          renderers={customRenderers.map(renderer => ({
```

Comment:
There is a problem with all inputs stores, or more particulary with the string and number input forms, that after you type one character, the input focus goes away and the next character you type goes to storybook, Then you click on the input field again and you can type one more character.

I think this is a tricky react problem.  Everytime you type a character, onChange gets called, whcih leads to setData getting called, which leads to this entire form getting regenerates, meaning all the input forms get recreated every time you type a character.   I think you have to code this in a way that renderers only gets generated once.  But I'm just guessing at this.  There might be something more in the docs or examples.  But the basic problem is you can't type more than one character in to the string or number field. 

Please see if you can figure it out through the docs.  But if you get stuck lets talk about it again.  This is an interesting problem. 

To see it in more detail, put some console.info statements in the   custominput render and you will see all the rerenders.